transaction/extra/src/memo/authenticated_common.rs

```diff
@@ -64,13 +30,13 @@ pub fn validate_authenticated_sender(
     let shared_secret =
         receiving_subaddress_view_private_key.key_exchange(sender_address.spend_public_key());
 
-    let expected_hmac = compute_category1_hmac(
+    let expected_hmac = mc_crypto_memo_mac::compute_category1_hmac(
         shared_secret.as_ref(),
         tx_out_public_key,
         memo_type_bytes,
-        memo_data,
+        &memo_data[..48].try_into().expect("length mismatch"),
```

Comment:
Can you say a little more about the slicing here?

Reply:
Thank you! Yes, I was curious about slicing `memo_data[..48]`, and the confidence level in `memo_data` being constructed correctly for that slice. It looks like the 16 bytes at the end will b the hmac, and we're fairly confident (plus catching the panic and raising `length mismatch` if there's an issue)